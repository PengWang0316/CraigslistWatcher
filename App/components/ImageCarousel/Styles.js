import { StyleSheet } from 'react-native';

export default StyleSheet.create({
  image: {
    width: 300,
    height: 250
  }
});
