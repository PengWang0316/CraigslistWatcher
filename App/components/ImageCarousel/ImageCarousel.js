import React, { Component } from 'react';
import Carousel from 'react-native-snap-carousel';
import PropTypes from 'prop-types';
import { Image, Dimensions } from 'react-native';

import Styles from './Styles';

/** Show a carousel for the item's images. */
class ImageCarousel extends Component {
  static propTypes = { images: PropTypes.array.isRequired };

  /**
   * Saving the carousel's reference.
   * @param {object} ref is the reference of this component.
   * @return {null} No return.
   */
  refFunction = ref => { this._carousel = ref; };

  /**
   * Creating the component to the Carousel.
   * @param {object} item is the image's url.
   * @return {jsx} No return the component's jsx.
   */
  renderItem = ({ item, index }) =>
    <Image style={Styles.image} source={{ uri: item }} key={index} />;

  /**
   * Rendering the jsx for the component.
   * @return {jsx} Return jsx.
   */
  render() {
    const { width } = Dimensions.get('window');
    return (
      <Carousel
        ref={this.refFunction}
        data={this.props.images}
        renderItem={this.renderItem}
        itemWidth={width - 100}
        sliderWidth={width}
        loop
      />);
  }
}
export default ImageCarousel;
