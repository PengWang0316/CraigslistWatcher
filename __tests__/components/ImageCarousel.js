import React from 'react';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';
import { Image } from 'react-native';

import ImageCarousel from '../../App/components/ImageCarousel';

// jest.mock('react-native', () => ({ Image: 'Image' }));
jest.mock('react-native-snap-carousel', () => 'Carousel');

describe('Test ImageCarousel', () => {
  const defaultProps = {
    images: ['image 1 url']
  };

  const getShallowComponent = (props = defaultProps) =>
    shallow(<ImageCarousel {...defaultProps} />);

  test('refFunction', () => {
    const component = getShallowComponent();
    component.instance().refFunction(component);
    expect(component.instance()._carousel).toBe(component);
  });

  test('renderItem', () => {
    const component = getShallowComponent();
    expect(component.instance().renderItem({ item: 'url', index: 1 })).toEqual(<Image style={{ height: 250, width: 300 }} source={{ uri: 'url' }} key={1} />);
  });

  test('Snapshot', () => expect(renderer.create(<ImageCarousel {...defaultProps} />).toJSON()).toMatchSnapshot());
});
