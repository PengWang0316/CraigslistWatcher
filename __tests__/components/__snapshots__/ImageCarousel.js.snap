// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test ImageCarousel Snapshot 1`] = `
<Carousel
  data={
    Array [
      "image 1 url",
    ]
  }
  itemWidth={650}
  loop={true}
  renderItem={[Function]}
  sliderWidth={750}
/>
`;
